src/general/MarkdownToBbCodeConverter.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Text.RegularExpressions;
+
+/// <summary>
+///   Converts markdown to bbcode for displaying
+/// </summary>
+public static class MarkdownToBbCodeConverter
+{
+    private static readonly Regex MarkdownLink = new(@"\[([^\]]+)\]\((http[^\)]+)\)", RegexOptions.Multiline);
+
+    private static readonly Regex BasicHtmlLink =
+        new(@"<a\s+[^>]*href=\""(http\S+)\""[^>]*>\s*(.+)\s*</a\s*>", RegexOptions.Compiled | RegexOptions.Multiline);
+
+    private static readonly Regex UrlWithoutMarkup = new(@"[^=](https?:\/\/[^\s)\]}]+)");
```

Comment:
This RegEx is faulty, it consumes the left parenthesis (`[^=]`).

One solution is to change this to `@"(^=)(https?:\/\/[^\s)\]}]+)"` and the conversion to `$"$1{Constants.CLICKABLE_TEXT_BBCODE}[url=$2]$2[/url]{Constants.CLICKABLE_TEXT_BBCODE_END}"`

Reply:
Yeah, I realized this based on your previous comment and basically ended up doing this exact change without seeing your comment first.